### PR TITLE
Bug 1683707: Add foreground count metric

### PR DESCRIPTION
**Pull request by mdboom**

### CODE REVIEW

**Review by chutten — COMMENTED** [review]
Any tests?

**Comment by mdboom** [comment]
> Any tests?

Oh, gee.  Of course, let me add some.

File: glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt [hunk 1]
```diff
@@ -192,6 +192,9 @@ class GleanTest {
             serverEndpoint = "http://" + server.hostName + ":" + server.port
         ))
 
+        // Send an "empty" metrics ping to reset the foreground_count
```
Comment (chutten) [inline glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt:195]: Oof.

**Review by chutten — APPROVED** [review]
Do we have to file a follow-up for adding a swift test to cover the swift implementation?

**Comment by mdboom** [comment]
> Do we have to file a follow-up for adding a swift test to cover the swift implementation?

We can do that, unless one of our Mac devs has a few cycles to port the Kotlin test to Swift in this PR.  (Cc: @badboy, @brizental, @travis79)

**Comment by brizental** [comment]
> We can do that, unless one of our Mac devs has a few cycles to port the Kotlin test to Swift in this PR.

Might be best to file a follow up because I don't think any of us can commit to your fork :) I'd be happy to take it as a follow up if necessary.

**Comment by mdboom** [comment]
> > We can do that, unless one of our Mac devs has a few cycles to port the Kotlin test to Swift in this PR.
> 
> Might be best to file a follow up because I don't think any of us can commit to your fork :) I'd be happy to take it as a follow up if necessary.

Fair enough.  I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1685712 for the follow-on work, and decided to remove all the Swift changes in this PR -- I think it better to commit the implementation and test at the same time.

File: glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt (hunk 1 above)
Comment (badboy) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt:195]: Shouldn't `resetGlean` take care of that? It has `clearStores = true`

Comment (mdboom) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt:195]: Yeah -- I really struggled to make this test work, and thought this was required, but it seems in the end it isn't.  Removing.

**Pull request merged**